src/context/store.ts

```diff
@@ -1,15 +1,67 @@
-import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
+import { Map } from 'immutable';
+import {
+  configureStore,
+  getDefaultMiddleware,
+  combineReducers,
+} from '@reduxjs/toolkit';
+import {
+  innerReducer,
+  outerReducer,
+  middleware,
+} from 'redux-async-initial-state';
+
 import mapStateReduce from './mapStateSlice';
+import serverStateReduce from './serverStateSlice';
+import { getLayersAvailableDates } from '../utils/server-utils';
+import { LAYERS_AVAILABLE_DATES } from '../constants';
+import { AvailableDates } from '../config/types';
 
-export const store = configureStore({
-  reducer: {
+const initializeStore = async (getCurrentState: any) => {
+  const isStorageSet = localStorage.getItem(LAYERS_AVAILABLE_DATES) !== null;
+
+  if (isStorageSet) {

```

Comment:
The dates change on the server as new data loads, I don't think this is a safe condition to have. We need to periodically update our view of available data.